scripts/release_scripts/check_vsa.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+# This script checks the Verification Summary Attestation generated by Macaron.
+
+# Strict bash options.
+#
+# -e:          exit immediately if a command fails (with non-zero return code),
+#              or if a function returns non-zero.
+#
+# -u:          treat unset variables and parameters as error when performing
+#              parameter expansion.
+#              In case a variable ${VAR} is unset but we still need to expand,
+#              use the syntax ${VAR:-} to expand it to an empty string.
+#
+# -o pipefail: set the return value of a pipeline to the value of the last
+#              (rightmost) command to exit with a non-zero status, or zero
+#              if all commands in the pipeline exit successfully.
+#
+# Reference: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html.
+set -euo pipefail
+
+# Log error (to stderr).
+log_err() {
+    echo "[ERROR]: $*" >&2
+}
+
+# Print usage message.
+print_help() {
+    cat << EOF
+Usage: $(basename "$0") [OPTIONS]
+
+This script checks the Verification Summary Attestation (VSA) generated by Macaron.
+
+Options:
+  --artifact-path PATH   Path to the artifact file.
+  --vsa-path PATH        Path to the Verification Summary Attestation (VSA) file.
+  --purl URL             Package URL to match against the VSA.
+  --help, -h             Display this help message and exit.
+
+Examples:
+  $(basename "$0") --artifact-path /path/to/artifact --vsa-path /path/to/vsa --purl "package-url"
+
+EOF
+}
+
+# Assert that a file exists.
+#
+# Arguments:
+#   $1: The path to the file.
+#   $2: The argument from which the file is passed into this script.
+#
+# With the `set -e` option turned on, this function exits the script with
+# return code 1 if the file does not exist.
+function assert_file_exists() {
+    if [[ ! -f "$1" ]]; then
+        log_err "File $1 does not exist."
+        return 1
+    fi
+}
+
+# Parse main arguments.
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --artifact-path)
+            arg_artifact_path="$2"
+            shift
+            ;;
+        --vsa-path)
+            arg_vsa_path="$2"
+            shift
+            ;;
+        --purl)
+            purl="$2"
+            shift
+            ;;
+        -h | --help)
+            print_help
+            exit 0
+            ;;
+        *)
+            log_err "Unknown option: $1"
+            print_help
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+
+# Check the artifact path and compute the checksum of the artifact.
+if [[ -n "${arg_artifact_path:-}" ]]; then
+    assert_file_exists "$arg_artifact_path"
+    artifact_checksum=$(shasum -a 256  "$arg_artifact_path" | awk '{print $1}')
```

Comment:
There is 2 spaces betwee `256` and `"$arg...`
I wonder if this is intended?

Reply:
See commit [32e104c](https://github.com/oracle/macaron/pull/858/commits/32e104c12ea49dec6675910dc6e9a3ea59baf4d9)